pkg/icons/icons.go

```diff
@@ -0,0 +1,144 @@
+package icons
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+const (
+	partnerFilePath     = "index.yaml"
+	partnerDownloadPath = "assets/logos"
+)
+
+// PackageIconOverride is a struct to hold the package name, path, and icon to override
+type PackageIconOverride struct {
+	Name string
+	Path string
+	Icon string
+}
+
+// PackageIconList is a list of PackageIconOverride
+type PackageIconList []PackageIconOverride
+
+// PackageIconMap is a map of PackageIconOverride
+type PackageIconMap map[string]PackageIconOverride
+
+// ParsePackageToOverride will convert a Package from a PackageList to a PackageIcon for a PackageIconList
```

Comment:
ok

Reply:
Looks like the comment is out of date. For that matter, I don't think we need a comment here since the function is so simple. And now, I'm wondering if we need this function. The code might be more readable without it... your choice.

Reply:
Let's keep it. 
I will just improve the comment. 

Go official linter goes crazy if I have a public function without a comment.